staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -667,5 +668,5 @@ func compareResults(r reflect.Value, expectedVal string) (bool, error) {
 		return false, errors.New("jsonpath leads to a nested object or list which is not supported")
 	}
 	s := fmt.Sprintf("%v", r.Interface())
-	return strings.TrimSpace(s) == strings.TrimSpace(expectedVal), nil
+	return strings.TrimSpace(s) == strings.TrimSpace(expectedVal) ||  expectedVal == "", nil
```

Comment:
Does the code guarantee that `s` is not empty? When the value is not provided in the wait for jsonpath, the expectation is that wait successfully completes on _any non empty_ value.

Reply:
From what I have understood from the code, it does seem like `s` cannot be empty. However, it would be great if someone with experience could confirm this!
Have added a check for this though

Reply:
According to the [signature] it can be empty string in an event `r.Interface()` returns an empty string.

[signature]: https://pkg.go.dev/reflect#Value.Interface